File: src/Listas1/pratica/Questao05.java
package Listas1.pratica;

import java.util.Scanner;

public class Questao05 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        String[] produtos = new String[n];
        int[] quantidade = new int[n];
        //falta colocar a parte dos produtos que foi vendida
        int[] quantidadeVendidos = new int[n];

        for (int i = 0; i < n; i++) {
            produtos[i] = sc.next();
            quantidade[i] = sc.nextInt();
            sc.nextLine();
        }

        String produtoDesejado = null;
        int quantidadeDesejada = 0;

        while (true) {
            produtoDesejado = sc.next();
            if ("fim".equals(produtoDesejado)) {
                for (int i = 0; i < n; i++) {
                    System.out.printf("%s %d\n", produtos[i], quantidade[i]);
                }
                return;
            }

            boolean encontrouProduto = false;
            for (int i = 0; i < n; i++) {
                encontrouProduto = produtoDesejado.equals(produtos[i]);
                if (encontrouProduto) {
                    quantidadeDesejada = sc.nextInt();
                    System.out.println(produtoDesejado + " " + quantidadeDesejada);
                    quantidade[i] -= quantidadeDesejada;
                    break;
                }
            }

            if (!encontrouProduto) {
                System.out.println(produtoDesejado + " produto não encontrado");
            }
        }
    }
}

File: Listas C++/Lista 1/questao05.cpp
#include <iostream>
#include <vector>
#include <string>
#include <utility>

int main() {
    
    int n;
    std::cin >> n;

    std::vector<std::pair<std::string, int>> estoque(n);

    for (int i = 0; i < n; ++i) {
        std::cin >> estoque[i].first;
        std::cin >> estoque[i].second;
    }

    std::string produtoDesejado;
    int quantidadeDesejada = 0;

    while (std::cin >> produtoDesejado) {
        if (produtoDesejado == "fim") {
            for (int i = 0; i < n; ++i) {
                std::cout << estoque[i].first << " " << estoque[i].second << "\n";
            }
            return 0;
        }

        bool encontrouProduto = false;
        for (int i = 0; i < n; ++i) {
            if (produtoDesejado == estoque[i].first) {
                encontrouProduto = true;
                std::cin >> quantidadeDesejada;
                std::cout << produtoDesejado << " " << quantidadeDesejada << std::endl;
                
                estoque[i].second -= quantidadeDesejada;
                
                break;
            }
        }

        if (!encontrouProduto) {
            std::cout << produtoDesejado << " produto não encontrado" << std::endl;
        }
    }

    return 0;
}